crcmocks/rbac.py

```diff
@@ -1,33 +1,83 @@
+from copy import deepcopy
+
 from flask import jsonify
 from flask import Blueprint
+from flask import request
+
+from crcmocks.util import get_user_rh_identity
+from crcmocks.config import DEFAULT_PERMISSIONS
 
 
 blueprint = Blueprint("rbac", __name__)
 
-rbac_response = {
-    "meta": {"count": 30, "limit": 1000, "offset": 0},
+BASE_RBAC_RESPONSE = {
+    "meta": {"count": None, "limit": None, "offset": 0},
     "links": {
         "first": "/api/rbac/v1/access/?application=&format=json&limit=1000&offset=0",
         "next": None,
         "previous": None,
         "last": "/api/rbac/v1/access/?application=&format=json&limit=1000&offset=0",
     },
-    "data": [
-        {"resourceDefinitions": [], "permission": "advisor:*:*"},
-        {"resourceDefinitions": [], "permission": "inventory:*:*"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:*"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-    ],
+    "data": [],
 }
 
 
-# TODO: allow customizable rbac response per user
-@blueprint.route("/v1/access/")
+# TODO: support resource definitions?
+@blueprint.route("/v1/access/", methods=["GET"])
 def rbac_access():
+    application = request.args.get("application")
+    limit = int(request.args.get("limit", 100))
+    offset = int(request.args.get("offset", 0))
+    # get the identity header
+    identity_header = request.headers.get("X-Rh-Identity")
+
+    if not identity_header:
+        return "No identity header present in request\n", 401
+
+    # get the user
+    user, username, account_number = get_user_rh_identity(identity_header)
+
+    if not user:
+        return f"No user found in TinyDB with username:" \
+               f" {username} or account_number: {account_number}\n", 404
+    if len(user) > 1:
+        return f"Multiple users found with:" \
+               f" {username} or account_number: {account_number}\n", 406
+
+    # get the user permissions
+    if user[0].get("entitlements") is not None:
+        permissions = user[0]["permissions"].split(",")
+    else:
+        permissions = DEFAULT_PERMISSIONS.copy()
+
+    rbac_response = deepcopy(BASE_RBAC_RESPONSE)
+    # set links
+    rbac_response["links"]["first"] = f"/api/rbac/v1/access/?application={application or ''}" \
```

Comment:
We should test that we got an actual application object back before setting these two urls.

Reply:
Do you mean checking to see that the application is actually valid? Hmm now that you mention it I don't think c.rh.c actually does this. 

I think it's OK as is, since the `or` will handle the `null`/`None` case.

Reply:
But ultimately this rbac response will be returned. The `data` key will be right, but the links would be incorrect if any tests is parsing those. I meant just a small check that the `application` returned isn't `None`.

Reply:
Oh I see what you're saying. Won't the `link` still be correct though in the case where `application` is `None`? 

e.g.
```python
x = None
f"testing none string {x or None}"
Out[7]: 'testing none string None'
f"testing none string {x or ''}"
Out[8]: 'testing none string '
```

Reply:
I mean _technically_ it will be correct right? But I don't think the end result will actually work. I suppose that's OK though, since we can still split on that and have the empty string be a result. So I think we're good here.